### PR TITLE
Support youtube videos in RelatedPage

### PR DESCRIPTION
**Pull request by chriswhong**
1) This PR introduces Youtube support for `RelatedPage` videos.  

Before, both a `vimeoId` and and `vimeoThumbnail` prop were required to use this component, which included the additional overhead of hosting a thumbnail file in the consuming repo and importing it with `AppropriateImage`.

Now, all that is needed is a `youtubeId`, and the component will use public URLs to display a thumbnail image hosted on YouTube.

Vimeo ids are no longer supported, and any videos embedded using `RelatedPage` should also be discoverable on their own on YouTube.  This is part of a larger initiative to only host our developer videos on YouTube instead of both Vimeo and Youtube.

2) Adds a `VideoModal` component which is used by `RelatedPage` to open the specified video in a modal.  It can also be used on its own, for a future scenario when we need a video to play in a modal (e.g. adding video quick links on the docs landing page)

3) Updates mr-ui to `2.7.0` which includes a fix for the modals being obscured behind the fixed top nav.

### CODE REVIEW

**Review by AlexanderBelokon — CHANGES_REQUESTED** [review]
The changes are mostly good, but since we're touching this, I think we might as well update the styles for modern devices.

Also, if this is a docs platform, shouldn't it seamlessly support video from any hosting service, or at least from the ones we already implemented support for? 

That is to say I'd leave Vimeo support in if we might have any videos there for any reason, it's not that big of a maintenance burden to have the video embed understand two types of video ids I think 🙃

File: src/components/related-page/related-page.js [hunk 1]
```diff
@@ -70,10 +71,15 @@ class RelatedPage extends React.PureComponent {
       video: {
         label: 'video',
         color: 'purple',
-        image: props.vimeoThumbnail ? (
-          <VimeoThumbnail>{props.vimeoThumbnail}</VimeoThumbnail>
+        image: props.youtubeId ? (
+          <VideoThumbnail>
+            <img
+              src={`https://img.youtube.com/vi/${this.props.youtubeId}/maxresdefault.jpg`}
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply src/components/related-page/related-page.js:77]: We were using appropriateimages for thumbnails which requires manually adding an image to the codebase.  My goal is to simplify the friction of adding videos, so removing the image step will save a lot of time and effort.

File: src/components/video-modal/video-modal.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        margin="large"
+        onExit={closeModal}
+      >
+        <div className="py36">
+          <div style={{ padding: '62.5% 0 0 0', position: 'relative' }}>
+            <iframe
+              id="ytplayer"
+              type="text/html"
+              width="640"
+              height="360"
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply src/components/video-modal/video-modal.js]: I updated `VideoModal` to include dimensions based on the viewport, with a sensible max-width of 1140px.

File: src/components/video-modal/video-modal.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    const { youtubeId, closeModal, title } = this.props;
+    return (
+      <Modal
+        accessibleTitle={`Video: ${title}`}
+        padding="none"
+        margin="large"
+        onExit={closeModal}
+      >
+        <div className="py36">
+          <div style={{ padding: '62.5% 0 0 0', position: 'relative' }}>
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply src/components/video-modal/video-modal.js]: These inline styles were removed

File: src/components/video-modal/video-modal.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+              width="640"
+              height="360"
+              title={title}
+              src={`https://www.youtube.com/embed/${youtubeId}?autoplay=1&origin=http://example.com`}
+              style={{
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                width: '100%',
+                height: '100%'
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply src/components/video-modal/video-modal.js]: These styles were removed

**Review by AlexanderBelokon — APPROVED** [review]
Looks great! 🚢

**Pull request merged**